[TASK-5516] feat: seo optimizations

kushagrasarathe opened the pull request:
Changes done:

- made a `seo.config` to create generate metadata for respective pages
- updated titles, keywords and other details for metadata
- added configuration files to generate `robots.txt` and `sitemap.xml`

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic metadata generation for various pages, enhancing SEO and social media sharing capabilities.
	- Added a new function to generate a sitemap, improving site indexing.
	- Implemented a robots.txt configuration to control web crawler access.

- **Bug Fixes**
	- Removed unused imports and static metadata declarations, streamlining the codebase.

- **Documentation**
	- Updated metadata definitions across multiple pages for consistency and improved structure.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kushagrasarathe committed:
feat: update metadata for using generateMetadata method

kushagrasarathe committed:
feat: add robots.ts and sitemap.ts configs

kushagrasarathe committed:
feat: update landing page metadata

kushagrasarathe committed:
refactor: update seo config

kushagrasarathe committed:
Merge branch 'develop' of https://github.com/peanutprotocol/peanut-ui into feat/seo-optimization

kushagrasarathe committed:
fix: robots config paths + nitpicks

kushagrasarathe committed:
fix: linting error

Hugo0 commented on src/config/seo.config.ts:
```diff
@@ -0,0 +1,103 @@
+            initialScale: 1,
+            maximumScale: 1,
+            userScalable: false,
+        },
```
yeah @kushagrasarathe idg why in seo config we are setting viewport? doesn't this seem like the wrong place to do this, independent of if the values are correct or no?

Hugo0 commented on src/config/seo.config.ts:
```diff
@@ -0,0 +1,103 @@
+import { Metadata } from 'next'
+
+type Page = 'landing' | 'cashout' | 'send' | 'requestCreate' | 'terms' | 'privacy' | 'jobs'
```
thought: hmmm, this duplicates the page definition. Now we have these page definitions, but they're also defined by the folder layout in next js. What if we change the name of send to something else? Also, relevant, now we merged terms and privacy as you can see in this PR: https://github.com/peanutprotocol/peanut-ui/pull/584/files . However, this page didn't get updated.

Suggestion: can we think of a different way to define the pages? Frankly, I think the setup of having the metadata defined close to the page (e.g. in src/app/cashout/page.tsx or /metadata.tsx) is a better abstraction

Hugo0 commented on src/config/seo.config.ts:
```diff
@@ -0,0 +1,103 @@
+    landing: {
+        title: 'Peanut Protocol | Cross-Chain Payment Infrastructure',
+        description:
+            'Seamless cross-chain payment infrastructure for sending and receiving digital assets. Built for both developers and consumers to abstract away blockchain complexities with chain-agnostic transfers, stablecoin conversions, and fiat offramps.',
```
nicely done on the descriptions

Hugo0 reviewed (changes requested):
Overall:
- this is missing pages (e.g. /profile)
- I don't think seo.config.ts is a good abstraction
- I think we should organize the stuff in src/app a bit. I suggest src/app/(routes) where we keep all the routes, and then it would also be clearer for SEO.

kushagrasarathe committed:
Merge branch 'develop' of https://github.com/peanutprotocol/peanut-ui into feat/seo-optimization

kushagrasarathe committed:
fix: update structure to declare metadata within pages
- replaced `seo.config.ts` with new `metadata.ts` in `/app` dir
- removed privacy and terms metadata
- updated robots and sitemap config to remove terms, privacy and blog page

kushagrasarathe committed:
fix: remove metadata from terms and privacy pages
